cobalt/browser/metrics/cobalt_metrics_service_client.cc

```diff
@@ -0,0 +1,213 @@
+// Copyright 2025 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cobalt/browser/metrics/cobalt_metrics_service_client.h"
+
+#include "base/i18n/rtl.h"
+#include "base/notreached.h"
+#include "base/path_service.h"
+#include "base/version.h"
+#include "components/metrics/metrics_log_uploader.h"
+#include "components/metrics/metrics_service.h"
+#include "components/metrics/metrics_state_manager.cc"
+#include "components/prefs/in_memory_pref_store.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service_factory.h"
+#include "components/variations/synthetic_trial_registry.h"
+#include "content/public/common/content_paths.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "url/gurl.h"
+
+namespace cobalt {
+
+namespace {
+
+class CobaltMetricsLogUploader : public metrics::MetricsLogUploader {
+ public:
+  CobaltMetricsLogUploader() = default;
+  ~CobaltMetricsLogUploader() = default;
+
+  void UploadLog(const std::string& compressed_log_data,
+                 const std::string& log_hash,
+                 const std::string& log_signature,
+                 const metrics::ReportingInfo& reporting_info) {
+    // TODO(b/372559349): Add logic here to report the Blob to the WebApp.
+    NOTIMPLEMENTED();
+  }
+};
+
+}  // namespace
+
+CobaltMetricsServiceClient::CobaltMetricsServiceClient()
+    : synthetic_trial_registry_(new variations::SyntheticTrialRegistry) {
+  DETACH_FROM_THREAD(thread_checker_);
+
+  // TODO(b/372559349): Extract initialization into a method if any of the
+  // following calls can fail (likely).
+
+  // No need to make `pref_registry` a member, `pref_service_` will keep a
+  // reference to it.
+  auto pref_registry = base::MakeRefCounted<PrefRegistrySimple>();
+  metrics::MetricsService::RegisterPrefs(pref_registry.get());
+
+  PrefServiceFactory pref_service_factory;
+  pref_service_factory.set_user_prefs(
+      base::MakeRefCounted<InMemoryPrefStore>());
```

Comment:
What are the implications of using InMemoryPrefStore here? Will we lose persistence across app lifetimes? FWIW, we use InMemoryPrefStore in the Cobalt UMA impl and it's been fine as far as I can tell... I just don't know if we're missing out on anything here. If so, a TODO/bug is likely appropriate.